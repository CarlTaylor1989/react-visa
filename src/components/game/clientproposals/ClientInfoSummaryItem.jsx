import React from 'react';
import PropTypes from 'prop-types';
import { I18n, Translate } from 'react-redux-i18n';
import DifficultyRating from '../../generic/rating/DifficultyRating';
import clientData from '../../../config/clients';

import './ClientInfoSummaryItem.scss';

const ClientInfoSummaryItem = ({ client, fieldName, type }) => (
  <div className={`clientInfoSummaryItem ${type}`}>
    <Translate className="infoSummaryTitle" value={`clientprops.info.${fieldName}`} />
    {type !== 'difficulty' ? (
      <span
        dangerouslySetInnerHTML={{ // eslint-disable-line react/no-danger
          __html: I18n.t(`clientprops.clients.${client}.info.${fieldName}`)
        }}
        className="infoSummaryText"
      />
    ) : (
      <DifficultyRating currentRating={clientData[client].difficulty} />
    )}
  </div>
);

ClientInfoSummaryItem.defaultProps = {
  type: ''
};

ClientInfoSummaryItem.propTypes = {
  client: PropTypes.string.isRequired,
  fieldName: PropTypes.string.isRequired,
  type: PropTypes.string
};

export default ClientInfoSummaryItem;
