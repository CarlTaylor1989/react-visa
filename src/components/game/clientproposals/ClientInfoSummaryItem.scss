@import '../../../styles/mixins';
@import '../../../styles/variables';

.clientInfoSummaryItem {
  @include align-items(flex-start);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  &.difficulty {
    @include align-items(flex-end);
    margin-left: auto;
  }

  &:not(:first-child) {
    padding: 0 15px;
  }

  &:nth-child(-n + 4) {
    border-right: 1px solid $colour-white;
  }

  &:first-child {
    padding-right: 15px;
  }

  .infoSummaryTitle {
    padding-bottom: 2px;
  }

  .infoSummaryTitle,
  .infoSummaryText {
    font-size: 16px;
  }

  .infoSummaryText,
  .avatarName {
    font-weight: 700;
  }
}
