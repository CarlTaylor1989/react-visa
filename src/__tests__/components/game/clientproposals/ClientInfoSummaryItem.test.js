/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import ClientInfoSummaryItem from
  '../../../../components/game/clientproposals/ClientInfoSummaryItem';

let mockProps;
beforeEach(() => {
  mockProps = {
    fieldName: 'account',
    client: 'client1'
  };
});

test('renders the ClientInfoSummaryItem component', () => {
  const wrapper = shallow(<ClientInfoSummaryItem {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ClientInfoSummaryItem component with the difficulty type', () => {
  const wrapper = shallow(<ClientInfoSummaryItem {...mockProps} type="difficulty" />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
