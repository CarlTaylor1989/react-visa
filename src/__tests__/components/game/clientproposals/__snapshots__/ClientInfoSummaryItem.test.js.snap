// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ClientInfoSummaryItem component 1`] = `
<div
  className="clientInfoSummaryItem "
>
  <Translate
    className="infoSummaryTitle"
    tag="span"
    value="clientprops.info.account"
  />
  <span
    className="infoSummaryText"
    dangerouslySetInnerHTML={
      Object {
        "__html": "account",
      }
    }
  />
</div>
`;

exports[`renders the ClientInfoSummaryItem component with the difficulty type 1`] = `
<div
  className="clientInfoSummaryItem difficulty"
>
  <Translate
    className="infoSummaryTitle"
    tag="span"
    value="clientprops.info.account"
  />
  <DifficultyRating
    currentRating={1}
    numberOfStars={3}
  />
</div>
`;
